File: mpd-concursos-app-frontend/src/app/core/services/dashboard/user-dashboard.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable, BehaviorSubject, throwError } from 'rxjs';
import { map, catchError, tap } from 'rxjs/operators';
import { environment } from '../../../../environments/environment';
import { LoggingService } from '@core/services/logging/logging.service';

/**
 * Interfaces para los datos del dashboard de usuario
 */
export interface UserDeadline {
  id: string;
  type: 'INSCRIPTION' | 'DOCUMENTS' | 'EXAM' | 'RESULT';
  title: string;
  description: string;
  deadline: string; // ISO date string
  daysRemaining: number;
  priority: 'HIGH' | 'MEDIUM' | 'LOW';
  contestId: string;
  actionRequired: string;
  route: string;
  isUrgent: boolean;
  status: 'ACTIVE' | 'EXPIRED' | 'COMPLETED';
  contestTitle: string;
  contestDepartment: string;
  documentType?: string;
  examType?: string;
  hoursRemaining: number;
}

export interface ProfileStats {
  completionPercentage: number;
  totalFields: number;
  completedFields: number;
  pendingFields: number;
  hasProfileImage: boolean;
  hasBasicInfo: boolean;
  hasContactInfo: boolean;
  hasEducation: boolean;
  hasExperience: boolean;
  lastUpdated: string;
}

export interface InscriptionStats {
  totalInscriptions: number;
  activeInscriptions: number;
  completedInscriptions: number;
  pendingInscriptions: number;
  cancelledInscriptions: number;
  frozenInscriptions: number;
  byStatus: Record<string, number>;
  byContest: Record<string, number>;
}

export interface DocumentStats {
  totalDocuments: number;
  pendingDocuments: number;
  approvedDocuments: number;
  rejectedDocuments: number;
  expiredDocuments: number;
  byType: Record<string, number>;
  byStatus: Record<string, number>;
}

export interface ExamStats {
  availableExams: number;
  completedExams: number;
  pendingExams: number;
  passedExams: number;
  failedExams: number;
  averageScore: number;
  byStatus: Record<string, number>;
}

export interface ActivityStats {
  totalLogins: number;
  lastLogin: string;
  documentsUploaded: number;
  profileUpdates: number;
  contestsViewed: number;
  accountCreated: string;
  daysActive: number;
}

export interface UserStats {
  profileStats: ProfileStats;
  inscriptionStats: InscriptionStats;
  documentStats: DocumentStats;
  examStats: ExamStats;
  activityStats: ActivityStats;
}

/**
 * Servicio para gestionar los datos del dashboard específicos del usuario
 * Conecta con los nuevos endpoints del backend para obtener vencimientos y estadísticas reales
 */
@Injectable({
  providedIn: 'root'
})
export class UserDashboardService {
  private readonly apiUrl = `${environment.apiUrl}/dashboard/user`;
  private readonly LOG_TAG = 'UserDashboardService';

  // BehaviorSubjects para cache de datos
  private deadlinesSubject = new BehaviorSubject<UserDeadline[]>([]);
  private statsSubject = new BehaviorSubject<UserStats | null>(null);

  constructor(
    private http: HttpClient,
    private loggingService: LoggingService
  ) {
    this.loggingService.debug(`[${this.LOG_TAG}] Initializing UserDashboardService.`, undefined, this.LOG_TAG);
  }

  /**
   * Obtiene los vencimientos próximos del usuario
   * @param daysAhead Número de días hacia adelante para buscar vencimientos (por defecto 30)
   * @returns Observable con la lista de vencimientos
   */
  getUserDeadlines(daysAhead: number = 30): Observable<UserDeadline[]> {
    this.loggingService.info(`[${this.LOG_TAG}] Fetching user deadlines for ${daysAhead} days ahead.`, undefined, this.LOG_TAG);

    const params = new HttpParams().set('daysAhead', daysAhead.toString());

    return this.http.get<UserDeadline[]>(`${this.apiUrl}/deadlines`, { params }).pipe(
      tap(deadlines => {
        this.loggingService.info(`[${this.LOG_TAG}] Successfully fetched ${deadlines.length} deadlines.`, deadlines, this.LOG_TAG);
        this.deadlinesSubject.next(deadlines);
      }),
      catchError(error => {
        this.loggingService.error(`[${this.LOG_TAG}] Error fetching user deadlines:`, error, this.LOG_TAG);
        // En caso de error, devolver datos mock para que la UI no se rompa
        const mockDeadlines = this.getMockDeadlines();
        this.deadlinesSubject.next(mockDeadlines);
        return throwError(() => error);
      })
    );
  }

  /**
   * Obtiene solo los vencimientos urgentes del usuario (próximos 7 días)
   * @returns Observable con la lista de vencimientos urgentes
   */
  getUrgentDeadlines(): Observable<UserDeadline[]> {
    this.loggingService.info(`[${this.LOG_TAG}] Fetching urgent deadlines.`, undefined, this.LOG_TAG);

    return this.http.get<UserDeadline[]>(`${this.apiUrl}/deadlines/urgent`).pipe(
      tap(deadlines => {
        this.loggingService.info(`[${this.LOG_TAG}] Successfully fetched ${deadlines.length} urgent deadlines.`, deadlines, this.LOG_TAG);
      }),
      catchError(error => {
        this.loggingService.error(`[${this.LOG_TAG}] Error fetching urgent deadlines:`, error, this.LOG_TAG);
        return throwError(() => error);
      })
    );
  }

  /**
   * Obtiene vencimientos por tipo específico
   * @param type Tipo de vencimiento (INSCRIPTION, DOCUMENTS, EXAM, RESULT)
   * @returns Observable con la lista de vencimientos del tipo especificado
   */
  getDeadlinesByType(type: string): Observable<UserDeadline[]> {
    this.loggingService.info(`[${this.LOG_TAG}] Fetching deadlines by type: ${type}.`, undefined, this.LOG_TAG);

    return this.http.get<UserDeadline[]>(`${this.apiUrl}/deadlines/type/${type}`).pipe(
      tap(deadlines => {
        this.loggingService.info(`[${this.LOG_TAG}] Successfully fetched ${deadlines.length} deadlines of type ${type}.`, deadlines, this.LOG_TAG);
      }),
      catchError(error => {
        this.loggingService.error(`[${this.LOG_TAG}] Error fetching deadlines by type ${type}:`, error, this.LOG_TAG);
        return throwError(() => error);
      })
    );
  }

  /**
   * Obtiene las estadísticas completas del usuario
   * @returns Observable con las estadísticas del usuario
   */
  getUserStats(): Observable<UserStats> {
    this.loggingService.info(`[${this.LOG_TAG}] Fetching complete user stats.`, undefined, this.LOG_TAG);

    return this.http.get<UserStats>(`${this.apiUrl}/stats`).pipe(
      tap(stats => {
        this.loggingService.info(`[${this.LOG_TAG}] Successfully fetched user stats.`, stats, this.LOG_TAG);
        this.statsSubject.next(stats);
      }),
      catchError(error => {
        this.loggingService.error(`[${this.LOG_TAG}] Error fetching user stats:`, error, this.LOG_TAG);
        // En caso de error, devolver datos mock
        const mockStats = this.getMockStats();
        this.statsSubject.next(mockStats);
        return throwError(() => error);
      })
    );
  }

  /**
   * Obtiene solo las estadísticas del perfil del usuario
   * @returns Observable con las estadísticas del perfil
   */
  getProfileStats(): Observable<ProfileStats> {
    this.loggingService.info(`[${this.LOG_TAG}] Fetching profile stats.`, undefined, this.LOG_TAG);

    return this.http.get<ProfileStats>(`${this.apiUrl}/stats/profile`).pipe(
      tap(stats => {
        this.loggingService.info(`[${this.LOG_TAG}] Successfully fetched profile stats.`, stats, this.LOG_TAG);
      }),
      catchError(error => {
        this.loggingService.error(`[${this.LOG_TAG}] Error fetching profile stats:`, error, this.LOG_TAG);
        return throwError(() => error);
      })
    );
  }

  /**
   * Obtiene solo las estadísticas de inscripciones del usuario
   * @returns Observable con las estadísticas de inscripciones
   */
  getInscriptionStats(): Observable<InscriptionStats> {
    this.loggingService.info(`[${this.LOG_TAG}] Fetching inscription stats.`, undefined, this.LOG_TAG);

    return this.http.get<InscriptionStats>(`${this.apiUrl}/stats/inscriptions`).pipe(
      tap(stats => {
        this.loggingService.info(`[${this.LOG_TAG}] Successfully fetched inscription stats.`, stats, this.LOG_TAG);
      }),
      catchError(error => {
        this.loggingService.error(`[${this.LOG_TAG}] Error fetching inscription stats:`, error, this.LOG_TAG);
        return throwError(() => error);
      })
    );
  }

  /**
   * Observables para acceso a datos cacheados
   */
  get deadlines$(): Observable<UserDeadline[]> {
    return this.deadlinesSubject.asObservable();
  }

  get stats$(): Observable<UserStats | null> {
    return this.statsSubject.asObservable();
  }

  /**
   * Refresca todos los datos del dashboard
   */
  refreshDashboardData(): Observable<{ deadlines: UserDeadline[], stats: UserStats }> {
    this.loggingService.info(`[${this.LOG_TAG}] Refreshing all dashboard data.`, undefined, this.LOG_TAG);

    return new Observable(observer => {
      Promise.all([
        this.getUserDeadlines().toPromise(),
        this.getUserStats().toPromise()
      ]).then(([deadlines, stats]) => {
        observer.next({ deadlines: deadlines!, stats: stats! });
        observer.complete();
      }).catch(error => {
        this.loggingService.error(`[${this.LOG_TAG}] Error refreshing dashboard data:`, error, this.LOG_TAG);
        observer.error(error);
      });
    });
  }

  /**
   * Datos mock para fallback en caso de error
   */
  private getMockDeadlines(): UserDeadline[] {
    return [
      {
        id: 'mock-1',
        type: 'INSCRIPTION',
        title: 'Inscripción: Concurso de Ejemplo',
        description: 'Cierre de inscripciones',
        deadline: new Date(Date.now() + 5 * 24 * 60 * 60 * 1000).toISOString(),
        daysRemaining: 5,
        priority: 'MEDIUM',
        contestId: 'mock-contest',
        actionRequired: 'Completar inscripción',
        route: '/dashboard/concursos/mock-contest',
        isUrgent: false,
        status: 'ACTIVE',
        contestTitle: 'Concurso de Ejemplo',
        contestDepartment: 'Departamento de Ejemplo',
        hoursRemaining: 120
      }
    ];
  }

  private getMockStats(): UserStats {
    return {
      profileStats: {
        completionPercentage: 65,
        totalFields: 7,
        completedFields: 4,
        pendingFields: 3,
        hasProfileImage: false,
        hasBasicInfo: true,
        hasContactInfo: true,
        hasEducation: false,
        hasExperience: false,
        lastUpdated: new Date().toISOString()
      },
      inscriptionStats: {
        totalInscriptions: 2,
        activeInscriptions: 1,
        completedInscriptions: 0,
        pendingInscriptions: 1,
        cancelledInscriptions: 0,
        frozenInscriptions: 0,
        byStatus: { active: 1, pending: 1 },
        byContest: { 'Concurso Ejemplo': 1, 'Otro Concurso': 1 }
      },
      documentStats: {
        totalDocuments: 3,
        pendingDocuments: 2,
        approvedDocuments: 1,
        rejectedDocuments: 0,
        expiredDocuments: 0,
        byType: { 'DNI': 1, 'CV': 1, 'Título': 1 },
        byStatus: { pending: 2, approved: 1 }
      },
      examStats: {
        availableExams: 0,
        completedExams: 0,
        pendingExams: 0,
        passedExams: 0,
        failedExams: 0,
        averageScore: 0,
        byStatus: {}
      },
      activityStats: {
        totalLogins: 15,
        lastLogin: new Date(Date.now() - 2 * 60 * 60 * 1000).toISOString(),
        documentsUploaded: 3,
        profileUpdates: 2,
        contestsViewed: 5,
        accountCreated: new Date(Date.now() - 30 * 24 * 60 * 60 * 1000).toISOString(),
        daysActive: 25
      }
    };
  }
}

File: concurso-backend/src/main/java/ar/gov/mpd/concursobackend/dashboard/application/service/UserDashboardService.java
package ar.gov.mpd.concursobackend.dashboard.application.service;

import ar.gov.mpd.concursobackend.dashboard.application.dto.UserDeadlineResponse;
import ar.gov.mpd.concursobackend.dashboard.application.dto.UserStatsResponse;
import ar.gov.mpd.concursobackend.dashboard.application.port.in.GetUserDeadlinesUseCase;
import ar.gov.mpd.concursobackend.dashboard.application.port.in.GetUserStatsUseCase;
import ar.gov.mpd.concursobackend.dashboard.application.port.out.LoadUserDashboardDataPort;
import ar.gov.mpd.concursobackend.dashboard.domain.UserDashboardStats;
import ar.gov.mpd.concursobackend.dashboard.domain.UserDeadline;
import ar.gov.mpd.concursobackend.dashboard.infrastructure.cache.DashboardCacheService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

/**
 * Servicio de aplicación para el dashboard del usuario
 * Implementa los casos de uso de obtener vencimientos y estadísticas
 */
@Service
@RequiredArgsConstructor
@Slf4j
@Transactional(readOnly = true)
public class UserDashboardService implements GetUserDeadlinesUseCase, GetUserStatsUseCase {
    
    private final LoadUserDashboardDataPort loadUserDashboardDataPort;
    private final DashboardCacheService cacheService;
    
    @Override
    public List<UserDeadlineResponse> getUserDeadlines(Long userId, Integer daysAhead) {
        log.info("Obteniendo vencimientos del usuario {} para los próximos {} días", userId, daysAhead);

        if (daysAhead == null) {
            daysAhead = 30; // Por defecto 30 días
        }

        LocalDateTime fromDate = LocalDateTime.now();
        LocalDateTime toDate = fromDate.plusDays(daysAhead);

        // Generar clave de cache basada en parámetros
        String cacheKey = "all_" + daysAhead + "d";

        // Intentar obtener desde cache
        List<UserDeadline> cachedDeadlines = cacheService.getUserDeadlines(userId, cacheKey);
        if (cachedDeadlines != null) {
            log.debug("Vencimientos obtenidos desde cache para usuario {}", userId);
            return cachedDeadlines.stream()
                    .map(this::mapToDeadlineResponse)
                    .collect(Collectors.toList());
        }

        // Si no está en cache, cargar desde base de datos
        List<UserDeadline> deadlines = loadUserDashboardDataPort.loadUserDeadlines(userId, fromDate, toDate);

        // Almacenar en cache antes de mapear
        cacheService.putUserDeadlines(userId, cacheKey, deadlines);

        List<UserDeadlineResponse> response = deadlines.stream()
                .map(this::mapToDeadlineResponse)
                .sorted((d1, d2) -> {
                    // Ordenar por fecha de vencimiento
                    if (d1.getDeadline() == null && d2.getDeadline() == null) return 0;
                    if (d1.getDeadline() == null) return 1;
                    if (d2.getDeadline() == null) return -1;
                    return d1.getDeadline().compareTo(d2.getDeadline());
                })
                .collect(Collectors.toList());

        log.info("Se encontraron {} vencimientos para el usuario {} (almacenados en cache)", response.size(), userId);
        return response;
    }
    
    @Override
    public List<UserDeadlineResponse> getUrgentDeadlines(Long userId) {
        log.info("Obteniendo vencimientos urgentes del usuario {}", userId);
        return getUserDeadlines(userId, 7).stream()
                .filter(deadline -> deadline.getIsUrgent() != null && deadline.getIsUrgent())
                .collect(Collectors.toList());
    }
    
    @Override
    public List<UserDeadlineResponse> getDeadlinesByType(Long userId, String type) {
        log.info("Obteniendo vencimientos del tipo {} para el usuario {}", type, userId);
        
        LocalDateTime fromDate = LocalDateTime.now();
        LocalDateTime toDate = fromDate.plusDays(90); // 3 meses para tipos específicos
        
        List<UserDeadline> deadlines;
        
        switch (type.toUpperCase()) {
            case "INSCRIPTION":
                deadlines = loadUserDashboardDataPort.loadInscriptionDeadlines(userId, fromDate, toDate);
                break;
            case "DOCUMENTS":
                deadlines = loadUserDashboardDataPort.loadDocumentDeadlines(userId, fromDate, toDate);
                break;
            case "EXAM":
                deadlines = loadUserDashboardDataPort.loadExamDeadlines(userId, fromDate, toDate);
                break;
            default:
                deadlines = loadUserDashboardDataPort.loadUserDeadlines(userId, fromDate, toDate);
                deadlines = deadlines.stream()
                        .filter(d -> d.getType().getValue().equalsIgnoreCase(type))
                        .collect(Collectors.toList());
        }
        
        return deadlines.stream()
                .map(this::mapToDeadlineResponse)
                .sorted((d1, d2) -> {
                    if (d1.getDeadline() == null && d2.getDeadline() == null) return 0;
                    if (d1.getDeadline() == null) return 1;
                    if (d2.getDeadline() == null) return -1;
                    return d1.getDeadline().compareTo(d2.getDeadline());
                })
                .collect(Collectors.toList());
    }
    
    @Override
    public UserStatsResponse getUserStats(Long userId) {
        log.info("Obteniendo estadísticas completas del usuario {}", userId);

        // Intentar obtener desde cache
        UserDashboardStats cachedStats = cacheService.getUserStats(userId);
        UserDashboardStats stats;

        if (cachedStats != null) {
            log.debug("Estadísticas obtenidas desde cache para usuario {}", userId);
            stats = cachedStats;
        } else {
            // Si no está en cache, cargar desde base de datos
            stats = loadUserDashboardDataPort.loadUserStats(userId);

            // Almacenar en cache
            cacheService.putUserStats(userId, stats);
            log.debug("Estadísticas cargadas desde BD y almacenadas en cache para usuario {}", userId);
        }

        return UserStatsResponse.builder()
                .profileStats(mapToProfileStatsResponse(stats.getProfileStats()))
                .inscriptionStats(mapToInscriptionStatsResponse(stats.getInscriptionStats()))
                .documentStats(mapToDocumentStatsResponse(stats.getDocumentStats()))
                .examStats(mapToExamStatsResponse(stats.getExamStats()))
                .activityStats(mapToActivityStatsResponse(stats.getActivityStats()))
                .build();
    }
    
    @Override
    public UserStatsResponse.ProfileStats getProfileStats(Long userId) {
        log.info("Obteniendo estadísticas del perfil del usuario {}", userId);
        
        UserDashboardStats stats = loadUserDashboardDataPort.loadUserStats(userId);
        return mapToProfileStatsResponse(stats.getProfileStats());
    }
    
    @Override
    public UserStatsResponse.InscriptionStats getInscriptionStats(Long userId) {
        log.info("Obteniendo estadísticas de inscripciones del usuario {}", userId);
        
        UserDashboardStats stats = loadUserDashboardDataPort.loadUserStats(userId);
        return mapToInscriptionStatsResponse(stats.getInscriptionStats());
    }
    
    /**
     * Mapea UserDeadline a UserDeadlineResponse
     */
    private UserDeadlineResponse mapToDeadlineResponse(UserDeadline deadline) {
        return UserDeadlineResponse.builder()
                .id(deadline.getId())
                .type(deadline.getType().getValue())
                .title(deadline.getTitle())
                .description(deadline.getDescription())
                .deadline(deadline.getDeadline())
                .daysRemaining(deadline.getDaysRemaining())
                .priority(deadline.getPriority().getValue())
                .contestId(deadline.getContestId())
                .actionRequired(deadline.getActionRequired())
                .route(deadline.getRoute())
                .isUrgent(deadline.isUrgent())
                .status(deadline.getStatus().getValue())
                .contestTitle(deadline.getContestTitle())
                .contestDepartment(deadline.getContestDepartment())
                .documentType(deadline.getDocumentType())
                .examType(deadline.getExamType())
                .hoursRemaining(deadline.getHoursRemaining())
                .build();
    }
    
    /**
     * Mapea ProfileStats de dominio a DTO
     */
    private UserStatsResponse.ProfileStats mapToProfileStatsResponse(UserDashboardStats.ProfileStats profileStats) {
        if (profileStats == null) {
            return UserStatsResponse.ProfileStats.builder()
                    .completionPercentage(0)
                    .totalFields(0)
                    .completedFields(0)
                    .pendingFields(0)
                    .hasProfileImage(false)
                    .hasBasicInfo(false)
                    .hasContactInfo(false)
                    .hasEducation(false)
                    .hasExperience(false)
                    .build();
        }
        
        return UserStatsResponse.ProfileStats.builder()
                .completionPercentage(profileStats.getCompletionPercentage())
                .totalFields(profileStats.getTotalFields())
                .completedFields(profileStats.getCompletedFields())
                .pendingFields(profileStats.getPendingFields())
                .hasProfileImage(profileStats.getHasProfileImage())
                .hasBasicInfo(profileStats.getHasBasicInfo())
                .hasContactInfo(profileStats.getHasContactInfo())
                .hasEducation(profileStats.getHasEducation())
                .hasExperience(profileStats.getHasExperience())
                .lastUpdated(profileStats.getLastUpdated() != null ? profileStats.getLastUpdated().toString() : null)
                .build();
    }
    
    /**
     * Mapea InscriptionStats de dominio a DTO
     */
    private UserStatsResponse.InscriptionStats mapToInscriptionStatsResponse(UserDashboardStats.InscriptionStats inscriptionStats) {
        if (inscriptionStats == null) {
            return UserStatsResponse.InscriptionStats.builder()
                    .totalInscriptions(0)
                    .activeInscriptions(0)
                    .completedInscriptions(0)
                    .pendingInscriptions(0)
                    .cancelledInscriptions(0)
                    .frozenInscriptions(0)
                    .build();
        }
        
        return UserStatsResponse.InscriptionStats.builder()
                .totalInscriptions(inscriptionStats.getTotalInscriptions())
                .activeInscriptions(inscriptionStats.getActiveInscriptions())
                .completedInscriptions(inscriptionStats.getCompletedInscriptions())
                .pendingInscriptions(inscriptionStats.getPendingInscriptions())
                .cancelledInscriptions(inscriptionStats.getCancelledInscriptions())
                .frozenInscriptions(inscriptionStats.getFrozenInscriptions())
                .byStatus(inscriptionStats.getByStatus())
                .byContest(inscriptionStats.getByContest())
                .build();
    }
    
    /**
     * Mapea DocumentStats de dominio a DTO
     */
    private UserStatsResponse.DocumentStats mapToDocumentStatsResponse(UserDashboardStats.DocumentStats documentStats) {
        if (documentStats == null) {
            return UserStatsResponse.DocumentStats.builder()
                    .totalDocuments(0)
                    .pendingDocuments(0)
                    .approvedDocuments(0)
                    .rejectedDocuments(0)
                    .expiredDocuments(0)
                    .build();
        }
        
        return UserStatsResponse.DocumentStats.builder()
                .totalDocuments(documentStats.getTotalDocuments())
                .pendingDocuments(documentStats.getPendingDocuments())
                .approvedDocuments(documentStats.getApprovedDocuments())
                .rejectedDocuments(documentStats.getRejectedDocuments())
                .expiredDocuments(documentStats.getExpiredDocuments())
                .byType(documentStats.getByType())
                .byStatus(documentStats.getByStatus())
                .build();
    }
    
    /**
     * Mapea ExamStats de dominio a DTO
     */
    private UserStatsResponse.ExamStats mapToExamStatsResponse(UserDashboardStats.ExamStats examStats) {
        if (examStats == null) {
            return UserStatsResponse.ExamStats.builder()
                    .availableExams(0)
                    .completedExams(0)
                    .pendingExams(0)
                    .passedExams(0)
                    .failedExams(0)
                    .averageScore(0.0)
                    .build();
        }
        
        return UserStatsResponse.ExamStats.builder()
                .availableExams(examStats.getAvailableExams())
                .completedExams(examStats.getCompletedExams())
                .pendingExams(examStats.getPendingExams())
                .passedExams(examStats.getPassedExams())
                .failedExams(examStats.getFailedExams())
                .averageScore(examStats.getAverageScore())
                .byStatus(examStats.getByStatus())
                .build();
    }
    
    /**
     * Mapea ActivityStats de dominio a DTO
     */
    private UserStatsResponse.ActivityStats mapToActivityStatsResponse(UserDashboardStats.ActivityStats activityStats) {
        if (activityStats == null) {
            return UserStatsResponse.ActivityStats.builder()
                    .totalLogins(0)
                    .documentsUploaded(0)
                    .profileUpdates(0)
                    .contestsViewed(0)
                    .daysActive(0)
                    .build();
        }
        
        return UserStatsResponse.ActivityStats.builder()
                .totalLogins(activityStats.getTotalLogins())
                .lastLogin(activityStats.getLastLogin() != null ? activityStats.getLastLogin().toString() : null)
                .documentsUploaded(activityStats.getDocumentsUploaded())
                .profileUpdates(activityStats.getProfileUpdates())
                .contestsViewed(activityStats.getContestsViewed())
                .accountCreated(activityStats.getAccountCreated() != null ? activityStats.getAccountCreated().toString() : null)
                .daysActive(activityStats.getDaysActive())
                .build();
    }
}
